### PR TITLE
[C#] Add implicit conversion operators for cs.internal.Null<T>

### LINKED ISSUE #4719: [C#] Add implicit conversion operators to cs.internal.Nullable<T>
Currently, working with Haxe-generated C# code can be a bit inconvenient, particularly with optional function arguments (which are currently handled via `cs.internal.Nullable<T>`).
It would make sense to define implicit conversion operators for Nullable, so that primitive\null values assigned to it from C# code would be converted into the structure automatically.
Wrote a [blog post](http://yal.cc/csharp-implicit-conversion-from-null-to-struct/) on the matter.
It would seem that there would be no side effects to this, since Haxe code already works with `.value`\`.hasValue` directly and structs are non-nullable.
Can assemble a pull request if needed.

### PR DESCRIPTION
**Pull request by YellowAfterlife**
Closes #4719.
Also corrected indentation levels in classCode\functionCode to match up with those of the generated code around them.

### CODE REVIEW

File: std/cs/internal/Null.hx [hunk 1]
```diff
@@ -21,21 +21,33 @@
...
+				return new haxe.lang.Null<T>((T) obj, true);
+			}
+		}
+
+		public static implicit operator haxe.lang.Null<T>(T val) {
+			return new haxe.lang.Null<T>(val, true);
+		}
+
+		public static implicit operator T(haxe.lang.Null<T> obj) {
+			return obj.@value;
```
Comment (nadako) [inline std/cs/internal/Null.hx]: I wouldn't do that, it's not safe. With `System.Nullable<T>` you have to do `HasValue/Value`, let's keep the same behaviour?

Comment (YellowAfterlife) [inline std/cs/internal/Null.hx]: When assigning Null<T> to T, Haxe currently generates "t = nt.@value", meaning that it'll return the default value (same as this function) if it's nulled. Can exclude this part from PR.

Comment (nadako) [inline std/cs/internal/Null.hx]: Yeah, sadly that's how Haxe currently works with Null, but I'm not sure we should leak that to outer world.

**Comment by nadako** [comment]
I merged it manually and removed implicit conversion to `T` (see 4d686ae814b07dabdb5ae07bbb15b34417fd258b)

**Pull request closed without merging**